libs/store_helper/kernel/ebpf_registry_helper.c

```diff
@@ -9,11 +9,56 @@
 #include "ebpf_registry_helper.h"
 #include "ebpf_windows.h"
 
+#define EBPF_STORE_TAG 'OTSE'
+
 #define _EBPF_RESULT(x) (NT_SUCCESS(x) ? EBPF_SUCCESS : EBPF_FAILED)
 
 ebpf_store_key_t ebpf_store_root_key = NULL;
 const wchar_t* ebpf_store_root_sub_key = EBPF_ROOT_REGISTRY_PATH;
 
+wchar_t*
+ebpf_get_wstring_from_string(_In_ const char* text)
+{
+    NTSTATUS status;
+    ANSI_STRING ansi_string;
+    UNICODE_STRING unicode_string = {0};
+    uint32_t new_size;
+
+    RtlInitAnsiString(&ansi_string, text);
+    new_size = RtlAnsiStringToUnicodeSize(&ansi_string);
+
+    // Allocate memory for the unicode string.
+    wchar_t* return_string = ExAllocatePoolUninitialized(NonPagedPoolNx, new_size, EBPF_STORE_TAG);
+    if (return_string == NULL) {
+        status = STATUS_NO_MEMORY;
```

Comment:
Given that we are only returning a pointer and no error code, do you want to add an error trace here?

Reply:
(Applies to the other implementation below too).

Reply:
This lib will be part of nuget package and will be linked into a 3rd party driver. So we may not be able to add any trace here (unless this lib has its own tracelogging provider and registers for tracing).